components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -7239,25 +7386,22 @@ private void handleException(String msg, Throwable t) throws APIManagementExcept
                     }
                 }
 
-                setAssociatedAPIProducts(identifier, uriTemplates);
+                setAssociatedAPIProducts(uuid, uriTemplates);
             } catch (SQLException e) {
                 handleException("Failed to get URI Templates of API" + identifier, e);
             }
         }
         return new LinkedHashSet<>(uriTemplates.values());
     }
 
-    public Map<Integer, URITemplate> getURITemplatesOfAPIWithProductMapping(APIIdentifier identifier)
-            throws APIManagementException {
+    public Map<Integer, URITemplate> getURITemplatesOfAPIWithProductMapping(String uuid) throws APIManagementException {
 
         Map<Integer, URITemplate> uriTemplates = new LinkedHashMap<>();
         Map<Integer, Set<String>> scopeToURITemplateId = new HashMap<>();
         try (Connection conn = APIMgtDBUtil.getConnection();
              PreparedStatement ps =
                      conn.prepareStatement(SQLConstants.GET_URL_TEMPLATES_OF_API_WITH_PRODUCT_MAPPINGS_SQL)) {
-            ps.setString(1, APIUtil.replaceEmailDomainBack(identifier.getProviderName()));
-            ps.setString(2, identifier.getName());
-            ps.setString(3, identifier.getVersion());
+            ps.setString(1, uuid);
             try (ResultSet rs = ps.executeQuery()) {
```

Comment:
This is the ideal way (get rid of APIIdentifier and use uuid directly). We should try to do the same in other places too.
Let's try to do it in a separate round.